src/types.rs

```diff
@@ -161,6 +162,34 @@ impl Position {
     }
 }
 
+impl PartialOrd for Position {
+    fn partial_cmp(&self, other: &Position) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Position {
+    fn cmp(&self, other: &Position) -> Ordering {
+        if self.commit < other.commit {
+            return Ordering::Less;
+        }
+
+        if self.commit == other.commit && self.prepare < other.prepare {
+            return Ordering::Less;
+        }
+
+        if self.commit > other.commit {
+            return Ordering::Greater;
+        }
+
+        if self.commit == other.commit && self.prepare > other.prepare {
+            return Ordering::Greater;
+        }
+
+        Ordering::Equal
```

Comment:
If all of this can be reordered like below, as it seems to be, so this `Impl` is useless because it could [have been `derived`](https://doc.rust-lang.org/std/cmp/trait.Ord.html#derivable).

> it will produce a lexicographic ordering based on the top-to-bottom declaration order of the struct's members.

```rust
if self.commit < other.commit {
    return Ordering::Less;
}

if self.commit > other.commit {
    return Ordering::Greater;
}

if self.commit == other.commit && self.prepare < other.prepare {
    return Ordering::Less;
}

if self.commit == other.commit && self.prepare > other.prepare {
    return Ordering::Greater;
}

Ordering::Equal
```

Reply:
That's good news. Any chance to be sure that's really what happens? Because,  `Position` is a pretty big deal in this case, and I don't want to investigate weird bugs because the generated code didn't do what I've expected :smile:

Reply:
ok, so, you could probably simplify this function by using the [`Ordering::then` method](https://doc.rust-lang.org/std/cmp/enum.Ordering.html?search=#method.then):

```rust
self.commit.cmp(&other.commit).then(self.prepare.cmp(&other.prepare))
```

Reply:
That's nice! :+1: